### PR TITLE
feat(cli): add macos hardened runtime signing config option

### PR DESCRIPTION
**Pull request by thewh1teagle**
In `macOS`, when building `dmg` with `ad-hoc` signature, and the app depends on Frameworks (dynamic shared libraries, `dylib` files), they won't load and the app will crash with error message box.

I found that if we disable signing flag in `macOS` bundler passed from `tauri-cli`, it won't crash.
the flag which passed is `--option runtime` in [tooling/bundler/src/bundle/macos/sign.rs#L210](https://github.com/tauri-apps/tauri/blob/dev/tooling/bundler/src/bundle/macos/sign.rs#L210)

<details>
<summary>crash log</summary>

```
Termination Reason:    Namespace DYLD, Code 1 Library missing
Library not loaded: @rpath/libavcodec.60.dylib
Referenced from: <D007FD2C-8AD0-336A-944D-C186621D1CEF> /Volumes/VOLUME/vibe.app/Contents/MacOS/vibe
Reason: tried: '/Volumes/vibe/vibe.app/Contents/Frameworks/libavcodec.60.dylib' (code signature in <BEFA8967-6321-3361-9151-FF5EE424F751> '/Volumes/vibe/vibe.app/Contents/Frameworks/libavcodec.60.dylib' not valid for use in process: mapping process and mapped file (non-platform) have different Team IDs), '/Volumes/vibe/vibe.app/Contents/Frameworks/libavcodec.60.dylib' (code signature in <BEFA8967-6321-3361-9151-FF5EE424F751> '/Volumes/vibe/vibe.app/Contents/Frameworks/libavcodec.60.dylib' not valid for use in process: mapping process and mapped file (non-platform) have different Team IDs)
(terminated at launch; ignore backtrace)
```
</details>

<details>

<summary>man codesign</summary>

```
     -o, --options flag,...
             During signing, specifies a set of option flags to be embedded in
             the code signature. The value takes the form of a comma-separated
             list of names (with no spaces). Alternatively, a numeric value
             can be used to directly specify the option mask (CodeDirectory
             flag word). See OPTION FLAGS below.

Flags:
             runtime              Preserve the hardened runtime version (-o
                                  runtime flag, --runtime-version option)
                                  instead of overriding or deriving the
                                  version.
```

</details>

This PR introduce a way to control the flag through `tauri.conf.json`:
```json
  "bundle": {
    "macOS": {
      "signingIdentity": "-",
      "hardenedRuntime": false
    }
  }
```

I believe that macOS bundling process needs to be much more flexible. 
Currently I have no other option to control the final signing of the app, and tauri enforce this `runtime` flag, as a result the final binary can't run and I can't publish it for macOS (arm).

meanwhile for testing the app, it's possible to resign it:

```console
codesign --force --deep -s "-" /Applications/vibe.app
hdiutil create -format UDZO -srcfolder some_app.app signed.dmg
```

### CODE REVIEW

**Comment by thewh1teagle** [comment]
Just a gentle reminder: I'm currently unable to test my app build locally due to the damaged `dmg` release caused by the default `hardened` option. It's impacting my development process. Any chance we could prioritize merging the PR? Appreciate your attention to this matter!

**Comment by cismous2204** [comment]
Especially useful, can anyone review and merge it? Thank you very much!

File: core/tauri-utils/src/config.rs [hunk 1]
```diff
@@ -583,6 +592,7 @@ impl Default for MacConfig {
       minimum_system_version: minimum_system_version(),
       exception_domain: None,
       signing_identity: None,
+      hardened_runtime: hardened_runtime(),
```
Comment (FabianLars) [inline core/tauri-utils/src/config.rs]:
```suggestion
      hardened_runtime: true,
```

File: core/tauri-utils/src/config.rs [hunk 2]
```diff
@@ -565,6 +565,11 @@ pub struct MacConfig {
   /// Identity to use for code signing.
   #[serde(alias = "signing-identity")]
   pub signing_identity: Option<String>,
+  /// Whether the codesign should enable [hardened runtime] (for executables) or not.
+  ///
+  /// [hardened runtime]: <https://developer.apple.com/documentation/security/hardened_runtime>
+  #[serde(alias = "hardened-runtime", default = "hardened_runtime")]
```
Comment (FabianLars) [inline core/tauri-utils/src/config.rs]:
```suggestion
  #[serde(alias = "hardened-runtime", default = "default_true")]
```

Just re-use the default_true function from (defined at the top of the file)

File: core/tauri-utils/src/config.rs [hunk 3]
```diff
@@ -575,6 +580,10 @@ pub struct MacConfig {
   pub dmg: DmgConfig,
 }
 
+fn hardened_runtime() -> bool {
+  true
+}
+
```
Comment (FabianLars) [inline core/tauri-utils/src/config.rs]:
```suggestion
```
With the 2 other changes this can be removed again

**Review by FabianLars — CHANGES_REQUESTED** [review]
Thanks, just some smaller things and then it's good to go :)

**Comment by thewh1teagle** [comment]
Thanks for the review : - )
Edited as suggested

**Review by FabianLars — APPROVED** [review]

**Pull request merged**